src/virtual_machine.rs

```diff
@@ -230,32 +266,42 @@ impl VM {
                 let spec = args & 0b0000_0000_0011_1111;
 
                 debug_assert_eq!(spec, 0b0000_0000_0011_1111);
-                Opcode::OpNot { dr, sr }
+                Opcode::Not { dr, sr }
             }
             // RTI
-            0b1000 => {
-                todo!()
-            }
+            0b1000 => Opcode::Rti,
             // ST
             0b0011 => {
-                todo!()
+                let sr = (args & 0b0000_1110_0000_0000) >> 9;
+                let offset9 = args & 0b0000_0001_1111_1111;
+                let offset = sign_extend(offset9, 9);
+                Opcode::St { sr, offset }
             }
             // STI
             0b1011 => {
-                todo!()
+                let sr = (args & 0b0000_1110_0000_0000) >> 9;
+                let offset9 = (args & 0b0000_0001_1111_1111) >> 9;
```

Comment:
Are you sure you want to bitshift here?

Reply:
Damn! Nice catch dylan. That's what happens when you copy and paste